redis/cluster.py

```diff
@@ -0,0 +1,1249 @@
+import copy
+import random
+import socket
+import time
+import warnings
+
+from collections import OrderedDict
+from redis.client import CaseInsensitiveDict, Redis, parse_info, PubSub
+from redis.commands import ClusterCommands, DataAccessCommands, \
+    ClusterManagementCommands
+from redis.connection import ClusterParser, ConnectionPool, Encoder, parse_url
+from redis.crc import key_slot
+from redis.exceptions import (
+    AskError,
+    BusyLoadingError,
+    ClusterDownError,
+    ClusterError,
+    DataError,
+    MovedError,
+    RedisClusterException,
+    ResponseError,
+    SlotNotCoveredError,
+    TimeoutError,
+    TryAgainError,
+)
+from redis.utils import dict_merge, list_keys_to_dict, merge_result, \
+    str_if_bytes
+
+
+def get_node_name(host, port):
+    return '{0}:{1}'.format(host, port)
+
+
+def get_connection(redis_node, *args, **options):
+    return redis_node.connection or redis_node.connection_pool.get_connection(
+        args[0], **options
+    )
+
+
+def parse_pubsub_channels(command, res, **options):
+    """
+    Result callback, handles different return types
+    switchable by the `aggregate` flag.
+    """
+    aggregate = options.get('aggregate', True)
```

Comment:
if options.get('aggregate', True) is False:
    return res
return merge_result(command, res)

Reply:
it's completely removed on the next commits